File: May-Leetcoding-Challenge/3. Ransom Note/RansomNote.py
class Solution:
    def canConstruct(self, ransomNote: str, magazine: str) -> bool:
        ransom_set = set(ransomNote)
        #randomNote = "aaab" -> ransom_set = "ab". Now it contains all elements with count 1.
        
        for char in ransom_set:
            if ransomNote.count(char) > magazine.count(char):
                return False
        return True
    
        '''
        l = list(magazine)
        for char in ransomNote:
            if char in l:
                l.remove(char)
            else:
                return False
        return True
        ''''

File: May-Leetcoding-Challenge/3. Ransom Note/RansomNote.java
class Solution {
    public boolean canConstruct(String ransomNote, String magazine) 
    {
        Map<Character, Integer> freq = new HashMap<>();  //Dictionary in Python
        
        for(char m: magazine.toCharArray())     //put() & get() function in HashMap() -> .put(char,freq)
            if(freq.containsKey(m))
                freq.put(m, freq.get(m) + 1);
            else
                freq.put(m,1);
        
        for(char r: ransomNote.toCharArray())
            if(freq.containsKey(r))
                if(freq.get(r) > 1)
                    freq.put(r, freq.get(r) - 1);   //as we check remove it from magazine.
                else
                    freq.remove(r);
            else
                return false;
                
        return true;
    }
}